### DISCUSSION TITLE
Job builder not saving to the DB

### DISCUSSION THREAD

**Discussion by CrazyMoka** [quartznet/quartznet#1634] [Q&A] [answered]
Hello, 
New to quartz. I have it setup to use the MSSQL database. I have all the tables and it can access them.

This is my sample code.  I'm trying to create a job and instantly run it.

        public async Task<OkObjectResult> Run(string jobName = "test")
        {
            IScheduler scheduler = await factory.GetScheduler();

            //...check if a specific job exists
            IJobDetail job = JobBuilder.Create<TestJob>().WithIdentity("myJob", "group1").Build();
           
            //if yes then run that job
            await scheduler.TriggerJob(new JobKey("myJob"));

            return Ok("OK");
        }

However. I get this error:
Couldn't store trigger 'DEFAULT.MT_5023352674260363642' for 'DEFAULT.myJob' job: The job (DEFAULT.myJob) referenced by the trigger does not exist.

I'm assuming because its not saving to the DB or I'm missing a step.

Thank you.

**Comment by lahma** [comment]
You kinda left the the critical part out, how you are storing the job. If no trigger is referring to job, Quartz removes it from database (it's considered orphaned). But if you want the job to stick around even without triggers referring to it, you should mark the job durable, can be done with job builder's `StoreDurably()` method.

**Reply by CrazyMoka** [reply]
Hi @lahma 
I changed it to used StoreDurably() and it still has the same error.

JobPersistenceException: The job (DEFAULT.myJob) referenced by the trigger does not exist.

Quartz.Impl.AdoJobStore.JobStoreSupport.StoreTrigger(ConnectionAndTransactionHolder conn, IOperableTrigger newTrigger, IJobDetail job, bool replaceExisting, string state, bool forceState, bool recovering, CancellationToken cancellationToken)
JobPersistenceException: Couldn't store trigger 'DEFAULT.MT_4140537578363881731' for 'DEFAULT.myJob' job: The job (DEFAULT.myJob) referenced by the trigger does not exist.

**Reply by lahma** [reply]
And you've called `scheduler.AddJob()` to actually add it?

**Reply by CrazyMoka** [reply]
IJobDetail job = JobBuilder.Create<TestJob>().WithIdentity("myJob", "group1").StoreDurably(true).Build();


            await scheduler.AddJob(job); <----- this is giving me and error Must by definition be durable.  But Job is durable... So not understanding the error.

**Reply by CrazyMoka** [reply]
never mind, I read the error message wrong.

**Comment by CrazyMoka** [comment, marked as answer]
still same error message:
JobPersistenceException: The job (DEFAULT.myJob) referenced by the trigger does not exist.

Quartz.Impl.AdoJobStore.JobStoreSupport.StoreTrigger(ConnectionAndTransactionHolder conn, IOperableTrigger newTrigger, IJobDetail job, bool replaceExisting, string state, bool forceState, bool recovering, CancellationToken cancellationToken)
JobPersistenceException: Couldn't store trigger 'DEFAULT.MT_4140537578363881731' for 'DEFAULT.myJob' job: The job (DEFAULT.myJob) referenced by the trigger does not exist.

**Reply by lahma** [reply]
Your code example for job has group `group1` but the error states that your trigger is trying to find job with default group name?

**Reply by CrazyMoka** [reply]
I guess. the code above is all I have. Should I be doing something different with group1 in trigger call?

**Reply by lahma** [reply]
Try `await scheduler.TriggerJob(new JobKey("myJob", "group1"));` or even better, use the same key that the job has that you build, `await scheduler.TriggerJob(job.Key);`

**Reply by CrazyMoka** [reply]
Yeah. I got it working. Thanks for the help!
